.github/workflows/terraform_provider.yml

```diff
@@ -225,6 +226,29 @@ jobs:
     - run: go install github.com/golangci/golangci-lint/cmd/golangci-lint
     - run: golangci-lint run ./aws/...
 
+  goreleaser:
+    needs: [go_mod_download]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - uses: actions/cache@v2
+        continue-on-error: true
+        timeout-minutes: 2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-pkg-mod-${{ hashFiles('go.sum') }}
+      - name: goreleaser check
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          args: check
```

Comment:
FYI: GoReleaser will return non-zero exit code here when it detects deprecated options in the config, which is probably good to catch either way, but you might still want to `continue-on-error` in such case as the config may otherwise be valid and the next step can still check if it builds on all platforms.